api/base/middleware.py

```diff
@@ -155,4 +216,102 @@ def process_response(self, request, response):
                 path=settings.CSRF_COOKIE_PATH,
                 httponly=settings.CSRF_COOKIE_HTTPONLY,
             )
-        return response
+
+        return super(SloanOverrideWaffleMiddleware, self).process_response(request, response)
+
+    @staticmethod
+    def get_domain(url: str) -> str:
+        """
+        http://localhost:8000/preprints -> localhost
+        http://osf.io/preprints/... -> .osf.io
+
+        :param url:
+        :return:
+        """
+        if url.startswith('http://localhost:'):
+            return 'localhost'
+        else:
+            netloc = urlparse(url).netloc
+            if netloc.count('.') > 1:
+                netloc = '.'.join(netloc.split('.'))
```

Comment:
What does this do? It looks like you're splitting this and putting it back together the way it was before.

Reply:
Opps, this leftover from when it was stripping subdomain info, good catch